src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_drv.c

```diff
@@ -76,15 +76,30 @@ struct xocl_pci_funcs userpf_pci_ops = {
 void xocl_reset_notify(struct pci_dev *pdev, bool prepare)
 {
 	struct xocl_dev *xdev = pci_get_drvdata(pdev);
+	struct xocl_subdev_info mbox;
+	char mbox_priv = 1;
+	int ret;
 
 	xocl_info(&pdev->dev, "PCI reset NOTIFY, prepare %d", prepare);
 
 	if (prepare) {
-	xocl_mailbox_set(xdev, PRE_RST, NULL);
-	xocl_subdev_destroy_by_id(xdev, XOCL_SUBDEV_DMA);
+	ret = xocl_subdev_get_devinfo(xdev, XOCL_SUBDEV_MAILBOX, &mbox, NULL);
+	if (ret) {
+		xocl_err(&pdev->dev, "can not get mailbox subdev");
+		return;
+	}
+	xocl_subdev_destroy_all(xdev);
+	mbox.priv_data = &mbox_priv;
+	mbox.data_len = 1;
+	ret = xocl_subdev_create_one(xdev, &mbox);
```

Comment:
After we recreate mailbox subdevice, i don't see xocl re-register the callback function for processing incoming message. Does the board still work after hot reset?